### PR TITLE
Add /report endpoint

### PR DESCRIPTION
**Pull request by erikjohnston**

### CODE REVIEW

File: synapse/storage/schema/delta/32/reports.sql [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+CREATE TABLE event_reports(
```
Comment (NegativeMjark) [inline synapse/storage/schema/delta/32/reports.sql]: a timestamp and an incrementing ID might be useful here.

**Comment by NegativeMjark** [comment]
Other than adding an incrementing ID so the schema can be updated incrementally. LGTM

**Comment by erikjohnston** [comment]
@NegativeMjark Like so?

File: synapse/storage/schema/delta/32/reports.sql (hunk 1 above)
Comment (erikjohnston) [inline synapse/storage/schema/delta/32/reports.sql]: Done

File: synapse/storage/schema/delta/32/reports.sql [hunk 2]
```diff
@@ -15,6 +15,8 @@
 
 
 CREATE TABLE event_reports(
+    id BIGINT NOT NULL,
```
Comment (NegativeMjark) [inline synapse/storage/schema/delta/32/reports.sql]: Maybe make the id a primary key?

File: synapse/storage/schema/delta/32/reports.sql [hunk 3]
```diff
@@ -15,6 +15,8 @@
...
+    received_ts BIGINT NOT NULL,
```
Comment (NegativeMjark) [inline synapse/storage/schema/delta/32/reports.sql]: Maybe chuck an index on the received_ts? (Not the end of the world if you don't. I imagine it should be fairly easy to add one later)

Comment (erikjohnston) [inline synapse/storage/schema/delta/32/reports.sql]: I really don't see the benefit of adding an index we don't use? I don't even know how this data is going to be used.

**Comment by NegativeMjark** [comment]
LGTM.

We might want to only allow one report per event per reporter. but like so much else with this PR it can probably wait until later.

**Pull request merged**